src/workspace.coffee

```diff
@@ -391,6 +394,11 @@ class Workspace extends Model
   #
   # Returns a promise that resolves to the {TextEditor} for the file URI.
   open: (uri, options={}) ->
+    cancelOpen = false
+    cancel = -> cancelOpen = true
+    @emitter.emit('will-open', {uri, cancel, options})
+    return if cancelOpen
```

Comment:
Kind of tricky since the caller may be expecting things to actually be open. Should it resolve with a boolean now?